test/addons-napi/test_new_target/binding.c

```diff
@@ -0,0 +1,30 @@
+#include <node_api.h>
+#include "../common.h"
+
+napi_value NewClass(napi_env env, napi_callback_info info) {
+  napi_value newTargetArg;
+  NAPI_CALL(env, napi_get_new_target(env, info, &newTargetArg));
+  napi_value thisArg;
+  NAPI_CALL(env, napi_get_cb_info(env, info, NULL, NULL, &thisArg, NULL));
+  napi_value undefined;
+  NAPI_CALL(env, napi_get_undefined(env, &undefined));
+
+  // this !== new.target since we are being invoked through super()
+  bool result;
+  NAPI_CALL(env, napi_strict_equals(env, newTargetArg, thisArg, &result));
```

Comment:
Is it worth also having an additional test where new.target === this?

Reply:
Yeah, it probably is. That would be a regular (non-super) construction case. I'll add one.